src/platform/src/main/java/org/geoserver/platform/resource/Paths.java

```diff
@@ -102,7 +102,41 @@ public static String path(String... path) {
         }
         return toPath(names);
     }
-
+    
+    // runtime flag which, if true, throws an error for the WARN characters
+    static final boolean STRICT_PATH = 
+            Boolean.valueOf(System.getProperty("STRICT_PATH", "true"));
```

Comment:
As discussed on the mailing list, this should be false by default since this commit fixes a regression.
But I also understand the desire to have things be more strict moving forward. 
How about making this a property in GeoServerInfo, allowing the users to edit it, and making it "true" in the release data directory, thus enforcing the more strict validation in new installations, while preserving functionality in old ones?